src/sysctl.rs

```diff
@@ -18,37 +23,111 @@ pub(crate) fn peri_reset<'a>() -> &'a sysctl::PERI_RESET {
     &sysctl().peri_reset
 }
 
+/// Accept freq_in as the input frequency,
+/// and try to find a set of parameters (nr, od, nf),
+/// which results in a frequency as near as possible to freq
+/// note for a PLL:
+///   freq_out = freq_in / nr * nf / od
+/// The reason why we don't port the complex config algorithm from the
+/// official C language SDK is that doing floating number arithmetics
+/// efficiently in no_std rust now is currently not very convenient
```

Comment:
Main obstacle I faced with is none of the methods on f64 in `std` is available in `no_std` is avaliable here, see [this issue](https://github.com/rust-lang/rust/issues/50145). What we need to port the algorithm from C SDK is `floor`, `ceil`, `round` and `abs`.
Some alternatives I have considered:
- Use some software floating point library to provide it. This adds a dependency and might not very efficient.
- Use functions in `core::intrinsics`, this force the user to use nightly rust.
- Wrap some asm code directly. Which might be complex and definitly `unsafe`.